cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381/Internal.hs

```diff
@@ -288,6 +307,22 @@ withNewAffine_ = fmap fst . withNewAffine
 withNewAffine' :: BLS curve => (AffinePtr curve -> IO a) -> IO (Affine curve)
 withNewAffine' = fmap snd . withNewAffine
 
+withAffineVector :: NonEmpty.NonEmpty (Affine curve) -> (AffinePtrVector curve -> IO a) -> IO a
+withAffineVector affines go = do
+  let numAffines = NonEmpty.length affines
+      sizeReference = sizeOf (undefined :: Ptr ())
```

Comment:
I know that it doesn't get evaluated, but that `undefined` still makes me very uncomfortable.  It looks like there's no alternative though: I tried `Proxy` and GHC didn't like it.